c10/xpu/XPUStream.cpp

```diff
@@ -0,0 +1,285 @@
+#include <c10/util/CallOnce.h>
+#include <c10/util/irange.h>
+#include <c10/xpu/XPUException.h>
+#include <c10/xpu/XPUStream.h>
+
+#include <atomic>
+#include <deque>
+#include <mutex>
+#include <vector>
+
+namespace c10::xpu {
+namespace {
+
+// Global stream state and constants
+c10::once_flag init_flag;
+DeviceIndex num_gpus = -1;
+constexpr int kStreamsPerPoolBits = 5;
```

Comment:
Please specify the exact int type you want `int32_t` or `int64_t`. This can cause issues on esoteric platforms.
This is true in other places in the PR and there were already comments about this that were not addressed?

Reply:
Thanks for the suggestions, I will submit an additional PR to cover this.